src/databricks/labs/ucx/installer/workflows.py

```diff
@@ -576,6 +576,8 @@ def _job_cluster_spark_conf(self, cluster_key: str):
             return {"spark.sql.sources.parallelPartitionDiscovery.parallelism": "200"} | conf_from_installation
         return conf_from_installation
 
+    # Workflow creation might fail on an InternalError with no message
+    @retried(on=[InternalError], timeout=timedelta(minutes=2))
```

Comment:
Do you think it's worth narrowing the scope of this retry to just the `.create()` call below that triggers it?

Reply:
yeah, I was not sure. Would like to reach out to someone internally to figure out what the underlying issue is to make a better decision here. For visibility of the problem, narrowing would be better. Generally, a retry on an InternalError makes sense

What has your preference?

Reply:
To avoid ambiguity, I don't think the `InternalError` can be narrowed. However the scope of the retry can be narrowed from the entire method (via annotation) to just the call though that triggers it, with something like:

```python
logger.info(f"Creating new job configuration for step={step_name}")
# Workflow creation might fail on an InternalError with no message.
retry = retried(on=[InternalError], timeout=timedelta(minutes=2))
new_job = retry(self._ws.jobs.create)(**settings)
assert new_job.job_id is not None
```

For an ambiguous error such as the one we're trying to deal with I think I prefer narrow targeting because it makes it clear which specific code path is triggering the issue.